src/makeCalculator.js

```diff
@@ -4,7 +4,38 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  let result = 0;
+
+  return {
+    get result() {
+      return result;
+    },
+
+    add(number) {
+      result += number;
+    },
+    subtract(number) {
+      result -= number;
+    },
+    multiply(number) {
+      result *= number;
+    },
+    divide(number) {
+      result /= number;
+    },
+
+    reset() {
+      result = 0;
+
+      return this;
+    },
+
+    operate(callback, number) {
+      callback(number);
```

Comment:
It's a good practice to check if `callback` is a function before invoking it to avoid runtime errors. Consider adding a check like `if (typeof callback === 'function') { callback(number); }`.